src/hooks/useAnnouncements.tsx

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'react'
+
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+import flag from 'cozy-flags'
+import { useClient } from 'cozy-client'
+
+import {
+  Announcement,
+  AnnouncementsConfigFlag
+} from 'components/Announcements/types'
+import { filterData } from 'components/Announcements/helpers'
+import { useAnnouncementsSettings } from './useAnnouncementsSettings'
+
+interface UseAnnouncementsProps {
+  canBeDisplayed: boolean
+}
+
+const useAnnouncements = ({
+  canBeDisplayed
+}: UseAnnouncementsProps): {
+  fetchStatus: string
+  data: Announcement[]
+} => {
+  const { lang } = useI18n()
+  const client = useClient()
+
+  const [fetchStatus, setFetchStatus] = useState('pending')
+  const [rawData, setRawData] = useState<Announcement[] | null>(null)
+  const [filteredData, setFilteredData] = useState<Announcement[]>([])
+  const [hasBeenFiltered, setHasBeenFiltered] = useState(false)
+  const config = flag<AnnouncementsConfigFlag>('home.announcements')
+  const { values, save } = useAnnouncementsSettings()
+
+  useEffect(() => {
+    const fetchAnnouncements = async (): Promise<void> => {
+      setFetchStatus('loading')
+      try {
+        const resp = (await client?.stackClient.fetchJSON(
+          'GET',
+          `/remote/${config?.remoteDoctype}?lang=${lang}&channels=${config?.channels}`
+        )) as { data: Announcement[] }
```

Comment:
nit: As we are in full typescript here, it may be better to use typeguards here, instead of casting with `as`. So we can also check the real type instead of trusting the web API's theoretical output